elements/lisk-p2p/test/integration/p2p.ts

```diff
@@ -1349,47 +1308,186 @@ describe('Integration tests for P2P library', () => {
+				const firstP2PNode = p2pNodeList[0];
+				const nodePortToResponsesMap: any = {};
+
+				const expectedAverageRequestsPerNode =
+					TOTAL_REQUESTS / NETWORK_PEER_COUNT_WITH_LIMIT;
+				const expectedRequestsLowerBound = expectedAverageRequestsPerNode * 0.4;
+				const expectedRequestsUpperBound = expectedAverageRequestsPerNode * 1.6;
+
+				for (let i = 0; i < TOTAL_REQUESTS; i++) {
+					const response = await firstP2PNode.request({
+						procedure: 'foo',
+						data: i,
+					});
+					let resultData = response.data as any;
+					if (!nodePortToResponsesMap[resultData.nodePort]) {
+						nodePortToResponsesMap[resultData.nodePort] = [];
+					}
+					nodePortToResponsesMap[resultData.nodePort].push(resultData);
+				}
+
+				for (let requestsHandled of Object.values(
+					nodePortToResponsesMap,
+				) as any) {
+					expect(requestsHandled).to.be.an('array');
+					expect(requestsHandled.length).to.be.greaterThan(
+						expectedRequestsLowerBound,
+					);
+					expect(requestsHandled.length).to.be.lessThan(
+						expectedRequestsUpperBound,
+					);
+				}
+			});
+		});
+
+		describe('P2P.send', () => {
+			let collectedMessages: Array<any> = [];
+
+			beforeEach(() => {
+				collectedMessages = [];
+				for (let p2p of p2pNodeList) {
+					p2p.on('messageReceived', message => {
+						collectedMessages.push({
+							nodePort: p2p.nodeInfo.wsPort,
+							message,
+						});
+					});
+				}
+			});
+
+			it('should send a message to a peers; should reach peers with even distribution', async () => {
+				const TOTAL_SENDS = 100;
+				const firstP2PNode = p2pNodeList[0];
+				const nodePortToMessagesMap: any = {};
+
+				const expectedAverageMessagesPerNode = TOTAL_SENDS;
+				const expectedMessagesLowerBound = expectedAverageMessagesPerNode * 0.4;
+				const expectedMessagesUpperBound = expectedAverageMessagesPerNode * 1.6;
+
+				for (let i = 0; i < TOTAL_SENDS; i++) {
+					firstP2PNode.send({ event: 'bar', data: i });
+				}
+
+				await wait(100);
+
+				for (let receivedMessageData of collectedMessages) {
+					if (!nodePortToMessagesMap[receivedMessageData.nodePort]) {
+						nodePortToMessagesMap[receivedMessageData.nodePort] = [];
+					}
+					nodePortToMessagesMap[receivedMessageData.nodePort].push(
+						receivedMessageData,
+					);
+				}
+
+				for (let receivedMessages of Object.values(
+					nodePortToMessagesMap,
+				) as any) {
+					expect(receivedMessages).to.be.an('array');
+					expect(receivedMessages.length).to.be.greaterThan(
+						expectedMessagesLowerBound,
+					);
+					expect(receivedMessages.length).to.be.lessThan(
+						expectedMessagesUpperBound,
+					);
+				}
+			});
+
+			it('should receive a message in the correct format', async () => {
+				const firstP2PNode = p2pNodeList[0];
+				firstP2PNode.send({ event: 'bar', data: 'test' });
+
+				await wait(100);
+
+				expect(collectedMessages).to.be.an('array');
+				expect(collectedMessages.length).to.be.greaterThan(6);
```

Comment:
Where does `6` come from?

Reply:
The number of collected messages was sometimes 7,8 or 9. I don't know if this can be an issue. What do you think @jondubois?

Reply:
Reverted